src/test/kotlin/com/epam/ahnl/service/StudentServiceTest.kt

```diff
@@ -0,0 +1,134 @@
+package com.epam.ahnl.service
+
+import com.epam.ahnl.model.entity.Student
+import com.epam.ahnl.model.repository.StudentRepository
+import com.epam.ahnl.service.dto.StudentDto
+import com.epam.ahnl.service.dto.StudentView
+import com.epam.ahnl.service.exception.StudentNotFoundException
+import io.mockk.confirmVerified
+import io.mockk.every
+import io.mockk.MockKAnnotations
+import io.mockk.unmockkAll
+import io.mockk.verify
+import io.mockk.impl.annotations.InjectMockKs
+import io.mockk.impl.annotations.MockK
+import io.mockk.junit5.MockKExtension
+import org.junit.jupiter.api.AfterAll
+import org.junit.jupiter.api.assertThrows
+import org.junit.jupiter.api.BeforeAll
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.TestInstance
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.extension.ExtendWith
+import org.springframework.data.domain.Page
+import org.springframework.data.domain.PageImpl
+import org.springframework.data.domain.PageRequest
+import org.springframework.data.domain.Pageable
+import java.time.LocalDate
+import java.util.Optional
+
+@ExtendWith(MockKExtension::class)
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+internal class StudentServiceTest {
+    @MockK
+    lateinit var repository: StudentRepository
+
+    @InjectMockKs
+    var service: StudentService = StudentService()
+    private lateinit var student: Student
+    private lateinit var studentView: StudentView
+
+    @BeforeAll
+    fun setUp() {
+        MockKAnnotations.init(this)
+        student = Student("ID", "First name", "Last name",
+                LocalDate.of(2000, 1, 1), "ah",
+                Student.Address("Country", "City", "Street"))
+        studentView = StudentView("ID", "${student.firstName} ${student.lastName}",
+                student.birthDate, student.faculty, student.address)
+    }
+
+    @AfterAll
+    fun tearDown() {
+        unmockkAll()
+    }
+
+    @Test
+    fun findAll() {
+        val pageable: Pageable = PageRequest.of(0, 1)
+        val studentList = mutableListOf(student)
+        val studentPage: Page<Student> = PageImpl(studentList, pageable, 1)
+        every { repository.findAll(pageable) } returns studentPage
```

Comment:
split logical blocks
1st data preparate
2nd mocks behaviour
3rd method invokation
4th verification
don't forget to separate them with gap lines

Reply:
Modified